WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -1196,9 +1203,50 @@ protected void onResume() {
                 mSelectedSiteRepository.hasSelectedSite() && mBottomNav != null
                 && mBottomNav.getCurrentSelectedPage() == PageType.MY_SITE
         );
+
+        checkForInAppUpdate();
+
         mIsChangingConfiguration = false;
     }
 
+    private void checkForInAppUpdate() {
+        mInAppUpdateManager.checkForAppUpdate(this, mInAppUpdateListener);
+    }
+
+    @NonNull final IInAppUpdateListener mInAppUpdateListener = new IInAppUpdateListener() {
+        @Override public void onAppUpdatePending() {
+            /* do nothing */
+        }
+
+        @Override public void onAppUpdateDownloaded() {
+            popupSnackbarForCompleteUpdate();
+        }
+        @Override public void onAppUpdateStarted(int type) {
```

Comment:
❓ Do we need all these callbacks on the interface and the listener if we are not using them?

Reply:
Nice catch here @AjeshRPai ! I want to keep the callbacks in case we need them for a future "client". I proceeded with a small refactoring, providing default empty implementations, so each client will be able to implement only the callbacks they need.

Reply:
Thanks Makes sense